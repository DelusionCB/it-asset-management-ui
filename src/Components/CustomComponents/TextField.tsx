import * as React from 'react';
import {Col, FormGroup, Input, Label} from 'reactstrap';
import {useTranslation} from 'react-i18next';
import './index.scss'

interface TextFieldProps {
    label: string
    validation?: object
    value?: string | undefined
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
    type?: 'text' | 'file' | 'textarea'
    id: string
    placeholder: string
    disabled: boolean
    invalid: boolean
}

function TextField ({label, validation, value, onChange, type, id, placeholder, disabled = false, invalid = false}: TextFieldProps): JSX.Element {
    const {t} = useTranslation()
    return (
        <Col className='textfield'>
            <FormGroup>
                <Label for={id}>
                    {t(label)}
                </Label>
                <Input
                    id={id}
                    disabled={disabled}
                    invalid={invalid}
                    aria-invalid={invalid}
                    name={id}
                    placeholder={`${t(placeholder)}`}
                    type={type}
                    value={value}
                    onChange={(e) => { onChange(e); }}
                />
            </FormGroup>
        </Col>
    );
}

export default TextField;
