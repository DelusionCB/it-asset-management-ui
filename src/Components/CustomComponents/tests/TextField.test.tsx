import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import TextField from '../TextField';
import '@testing-library/jest-dom'

describe('<TextField  />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <TextField id={'id'} type='text' invalid={false} label='label' onChange={jest.fn()} placeholder={'placeholder'} disabled={false}/>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
