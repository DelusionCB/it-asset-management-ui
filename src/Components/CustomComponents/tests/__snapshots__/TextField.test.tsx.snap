// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField  /> Snapshot renders correctly 1`] = `
<div
  className="textfield col"
>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="id"
    >
      label
    </label>
    <input
      aria-invalid={false}
      className="form-control"
      disabled={false}
      id="id"
      name="id"
      onChange={[Function]}
      placeholder="placeholder"
      type="text"
    />
  </div>
</div>
`;
